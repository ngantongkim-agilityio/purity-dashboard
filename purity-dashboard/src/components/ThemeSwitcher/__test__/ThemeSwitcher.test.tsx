import { render } from '@testing-library/react';

// Components
import { ThemeSwitcher } from '..';

describe('ThemeSwitcher component', () => {
  beforeEach(() => {
    jest.mock('next-themes', () => ({
      useTheme: jest.fn(() => ({
        setTheme: jest.fn(),
        theme: 'light'
      }))
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render ThemeSwitcher component with default props', () => {
    const { container } = render(<ThemeSwitcher />);

    expect(container).toMatchSnapshot();
  });
});
