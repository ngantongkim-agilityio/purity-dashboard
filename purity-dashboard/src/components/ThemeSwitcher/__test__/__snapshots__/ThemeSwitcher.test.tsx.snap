// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher component Should render ThemeSwitcher component with default props 1`] = `
<div>
  <button
    aria-label="theme switcher"
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 !gap-0 transition-transform-colors-opacity motion-reduce:transition-none w-10 data-[hover=true]:opacity-hover border-transparent font-semibold p uppercase text-white bg-none px-4 min-w-20 h-9 text-xs gap-2 rounded-xl bg-transparent"
    type="button"
  >
    <svg
      class="h-5 w-5 text-primary"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 3v2.25m6.364.386-1.591 1.591M21 12h-2.25m-.386 6.364-1.591-1.591M12 18.75V21m-4.773-4.227-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
